safe_landing_planner/include/safe_landing_planner/grid.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+namespace avoidance {
+
+class Grid {
+public:
+  Grid(const float grid_size, const float cell_size) :
+    grid_size_(grid_size),
+    cell_size_(cell_size),
+    mean_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size))),
+    variance_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size))),
+    counter_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size))),
+    land_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size)))
+    {
+      reset();
+    }
+  ~Grid() = default;
+
+  void reset() {
+    mean_.fill(0.f);
+    variance_.fill(0.f);
+    counter_.fill(0);
+    land_.fill(0);
+  }
+
+  void resize() {
+    mean_.resize(static_cast<int>(std::ceil(grid_size_/cell_size_)), static_cast<int>(std::ceil(grid_size_/cell_size_)));
+    variance_.resize(static_cast<int>(std::ceil(grid_size_/cell_size_)), static_cast<int>(std::ceil(grid_size_/cell_size_)));
+    counter_.resize(static_cast<int>(std::ceil(grid_size_/cell_size_)), static_cast<int>(std::ceil(grid_size_/cell_size_)));
+    land_.resize(static_cast<int>(std::ceil(grid_size_/cell_size_)), static_cast<int>(std::ceil(grid_size_/cell_size_)));
+    reset();
+  }
+
+  void setMean(Eigen::Vector2i &idx, float value) {
```

Comment:
```suggestion
  void setMean(const Eigen::Vector2i &idx, float value) {
```

And similarly for 
`setVariance`
`increaseCounter`
`getMean`
`getVariance`
`getCounter`
`setFilterLimits`